api/app/db/models/auto_spatial_advisory.py

```diff
@@ -175,20 +171,38 @@ class AdvisoryElevationStats(Base):
 
 
 class AdvisoryFuelStats(Base):
-    """ 
+    """
     Summary statistics about the fuel type of area with high hfi (4k-10k and >10k) per fire shape
     based on the set run_type, for_date and run_datetime.
     """
-    __tablename__ = 'advisory_fuel_stats'
-    __table_args__ = (
-        UniqueConstraint('advisory_shape_id', 'threshold', 'run_parameters', 'fuel_type'),
-        {
-            'comment': 'Fuel type stats per fire shape by advisory threshold'
-        }
-    )
+
+    __tablename__ = "advisory_fuel_stats"
+    __table_args__ = (UniqueConstraint("advisory_shape_id", "threshold", "run_parameters", "fuel_type"), {"comment": "Fuel type stats per fire shape by advisory threshold"})
     id = Column(Integer, primary_key=True, index=True)
     advisory_shape_id = Column(Integer, ForeignKey(Shape.id), nullable=False, index=True)
     threshold = Column(Integer, ForeignKey(HfiClassificationThreshold.id), nullable=False)
     run_parameters = Column(Integer, ForeignKey(RunParameters.id), nullable=False, index=True)
     fuel_type = Column(Integer, ForeignKey(SFMSFuelType.id), nullable=False, index=True)
     area = Column(Float, nullable=False)
+
+
+class AdvisoryTPIStats(Base):
+    """
+    Summary statistics about the elevation based on a classified Topographic Position Index (TPI).
+    Classification of the TPI are bucketed into 1 = valley bottom, 2 = mid slope, 3 = upper slope.
+    Each record includes the raster pixel count of the above classifications as well as the raster pixel size and resolution in metres
+    and an advisory shape the stats are related to.
+    """
+
+    __tablename__ = "advisory_tpi_stats"
+    __table_args__ = {"comment": "Elevation TPI stats per fire shape"}
+    id = Column(Integer, primary_key=True, index=True)
+    advisory_shape_id = Column(Integer, ForeignKey(Shape.id), nullable=False, index=True)
+    threshold = Column(Integer, ForeignKey(HfiClassificationThreshold.id), nullable=False)
+    run_parameters = Column(Integer, ForeignKey(RunParameters.id), nullable=False, index=True)
+    valley_bottom = Column(Integer, nullable=False, index=True)
+    mid_slope = Column(Integer, nullable=False, index=True)
+    upper_slope = Column(Integer, nullable=False, index=True)
+    raster_height = Column(Integer, nullable=False, index=True)
```

Comment:
What do `raster_height` and `raster_width` represent?

Reply:
I took the wording from QGIS here, I think they represent # of cells in column and row of the raster
![wdithheight](https://github.com/user-attachments/assets/6324dfba-dcc0-4492-b285-809c266e8256)

Reply:
I'm wondering if we need these values? They relate to the underling raster, not to the fire zone unit we're storing data for so I'm not clear on the potential value they provide.

Reply:
Ah good point, added the table before I had broken zone the raster by fire zone. I'll delete it!